### ISSUE TITLE
Replay crashes on Pixel 2

### ISSUE THREAD

**Issue by Qining** [google/gapid#1246]

**Comment by Qining** [comment]
Crashes at VkBindImage, confirmed not related to DCE.

**Comment by Qining** [comment]
bloom works fine, but cube: (https://github.com/google/vulkan_test_applications/tree/master/application_sandbox/cube) crashes on replay.

**Comment by Qining** [comment]
Apps from https://github.com/SaschaWillems/Vulkan works fine, except that Pixel 2 Vulkan does not support geometry shader and tessellation shader in the current driver. Besides the coherent memory probably is not working as expected: some textures missing. But the framebuffers from original app run/tracing/replay are the same.

Apps from https://github.com/google/vulkan_test_applications does not work due to "Out of Memory" on replay, specifically, when `vkAllocateMemory` is called with a request of 512MB of device local memory. Changing the app's memory requirements fixes the problem. However, the problem now is that, tracing works as expected, but replay fails, for a full trace start at the beginning, but why the replay hits the "Out of Memory" but not the tracing?

Error message:
```
- Adreno-GSL: <sharedmem_gpuobj_alloc:2318>: sharedmem_gpumem_alloc: mmap failed errno 12 Out of Memory
- Adreno-GSL: <gsl_memory_alloc_pure: 2190>: GSL MEM ERROR: kgsl_sharedmem_alloc ioctl failed.
```

**Comment by Qining** [comment]
Change to low priority as this probably only happens when requesting a very large memory space.

**Comment by Qining** [comment]
Fixed with commit: 157c3c7651cf0032010965be5dac7046f0363827
